### PR TITLE
Feat: update existing document #52

### PR DESCRIPTION
**Pull request by spoxies**

### CODE REVIEW

**Comment by spoxies** [comment]
Hangt tight... Tests fail because I made a weird/merge/copy mistake

**Comment by MichaelSolati** [comment]
It's cool, just let me know whenever you're ready so I can start checking it out

**Comment by spoxies** [comment]
I suspect that in this case The Travis build failed because (most likely), an other build was/is running. @MichaelSolati would you be able/willing to trigger/start an other build?

**Comment by MichaelSolati** [comment]
Looks like it passed, I'll review it over the next few days!

File: src/geofirestore.ts [hunk 1]
```diff
@@ -96,21 +96,30 @@ export class GeoFirestore {
...
    * If any provided key already exists in this GeoFirestore, it will be overwritten with the new location value.
    *
+   * @param writeType Specifies the type of write (set || update).
    * @param keyOrDocuments The key representing the document to add or an object of $key - document pairs.
    * @param document The document to be added to the GeoFirestore.
    * @param customKey The key of the document to use as the location. Otherwise we default to `coordinates`.
    * @returns A promise that is fulfilled when the write is complete.
    */
-  public set(keyOrDocuments: string | any, document?: any, customKey?: string): Promise<any> {
+  private write(writeType: string, keyOrDocuments: string | any, document?: any, customKey?: string): Promise<any> {
```
Comment (MichaelSolati) [inline src/geofirestore.ts]: Please ensure all private functions are underscored => `private _write`.

Comment (MichaelSolati) [inline src/geofirestore.ts]: Also private functions and public functions are kept separate, and in alphabetical order. I'm not sure if this is a change needed anywhere else, but please keep that in mind (and update the code to reflect that).

File: src/utils.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { firestore, GeoFirestoreObj, QueryCriteria } from './interfaces';
+import { triggerAsyncId } from 'async_hooks';
```
Comment (MichaelSolati) [inline src/utils.ts]: What is this used for?

File: test/common.ts [hunk 3]
```diff
@@ -137,6 +137,7 @@ export class Checklist {
  * went wrong.
  **/
 export function failTestOnCaughtError(error) {
+  console.log(error);
```
Comment (MichaelSolati) [inline test/common.ts]: Please remove all console logs

File: src/utils.ts (hunk 2 above)
Comment (spoxies) [reply src/utils.ts]: Thank you! Removed in https://github.com/geofirestore/geofirestore-js/pull/53/commits/324aa9604d6e9419711eac77d1014706eafe3791

Probably my (overly) helpful editor tough this project needed this.

File: src/geofirestore.ts (hunk 1 above)
Comment (spoxies) [reply src/geofirestore.ts]: Good call! 
Corrected in : https://github.com/geofirestore/geofirestore-js/pull/53/commits/716eda9ed356cb1a078f6fc9fe1385b37973c6af

Comment (spoxies) [reply src/geofirestore.ts]: Corrected in : https://github.com/geofirestore/geofirestore-js/pull/53/commits/716eda9ed356cb1a078f6fc9fe1385b37973c6af

**Comment by MichaelSolati** [comment]
I'm gonna bring it into a branch so that I can work on it, pretty much it's just failing one test. Once I figure that out and resolve it I'll bring it into the main branch and this will be version 2.3.0

**Comment by MichaelSolati** [comment]
Or I guess it works when I merged it in... 🙃

ANYWAY, a few more things I wanna do, then it'll come into dev/master. I''ll use the issue you opened for all future communications about this feat.

**Comment by spoxies** [comment]
> Or I guess it works when I merged it in...
@MichaelSolati  hahaha yes, you have to be a bit lucky to have the test succeed . When an other person/system runs the test in that same period, the check fails. 

Good luck and thank you!

**Pull request merged**